### DISCUSSION TITLE
dynamic values in 'it' block description

### DISCUSSION THREAD

**Discussion by eriksabol** [cypress-io/cypress#15855] [Questions and Help] [answered]
Hi all, 

I need to display dynamically generated string from an async request in the description of the ‘it’ block.

`it('my description of ' + asyncReturnedValue, () => {
do stuff
    })`

I tried before() hook placed inside support file, however the overall workflow is initialized before the async value is returned. Code from inside before() hook is implicitly executed inside BEFORE ALL as part of the first test in the row. As a result, I see undefined for my string in the ‘it’ block description (which is correct but not desired). 

From the documentation:
> Cypress executes the support file before the spec file. For example when you click on a test file named spec-a.js via cypress open, then the Test Runner executes the files in the following order:
> ```
> 
> <script src="support/index.js"></script>
> 
> <script src="integration/spec-a.js"></script>
> ```

What is happening before the support file is executed? Is there a way to somehow delay initialization of a workflow to wait for certain value? I know I could generate the async value outside cypress and pass it via env.variable,.. However, I wonder if I could do it from within cypress.

Any input very much appreciated, thanks.

**Comment by bahmutov** [comment, marked as answer]
Unfortunately, there is no way to execute some code in the spec file, then generate the tests, and then start running them. You have to generate all the values to produce the tests before the spec starts, and the plugin file is the only place that I think would work for you. See the "Dynamic tests" recipes in https://github.com/cypress-io/cypress-example-recipes#fundamentals with live examples

**Reply by eriksabol** [reply]
Hi Gleb,

thanks for the dynamic tests link! I literally took my computer and coded it inside plugins immediately, the best part is that it really WORKS! :) Thanks again.

Erik.
